src/sdl/index.ts

```diff
@@ -141,6 +137,25 @@ export class SDL {
     }
   }
 
+  constructor(
+    public readonly data: BetaSdl<V>,
+    public readonly version: NetworkVersion = "beta2",
+    public readonly network: NetworkId = "mainnet"
+  ) {
+    this.validate();
+  }
+
+  private validate() {
```

Comment:
having an instance method allows to run validation on instance initialisation uniformly. It would also allow to use `this` down the nested validation calls instead of long args passage along call stacks.